File: Algorithms/2660-Determine-the-Winner-of-a-Bowling-Game/Solution.py
class Solution(object):
    def isWinner(self, player1, player2):
        """
        :type player1: List[int]
        :type player2: List[int]
        :rtype: int
        """
        s1, s2, pp1, pp2 = player1[0], player2[0], -1, -1
        
        for i in range(1, len(player1)):
            s1 += (2 * player1[i]) if (player1[i - 1] == 10 or pp1 == 10) else player1[i]
            s2 += (2 * player2[i]) if (player2[i - 1] == 10 or pp2 == 10) else player2[i]
            pp1, pp2 = player1[i - 1], player2[i - 1]

        return 0 if s1 == s2 else 1 if s1 > s2 else 2
    

File: Algorithms/2660-Determine-the-Winner-of-a-Bowling-Game/Solution.js
/**
 * @param {number[]} player1
 * @param {number[]} player2
 * @return {number}
 */
var isWinner = function(player1, player2) {
    let s1 = player1[0], s2 = player2[0], i = 1,
        p1 = player1[i - 1], p2 = player2[i - 1],
        pp1 = -1, pp2 = -1;
    
    while(i < player1.length) {
        s1 += isTen(p1, pp1) ? 2 * player1[i] : player1[i];
        s2 += isTen(p2, pp2) ? 2 * player2[i] : player2[i];

        pp1 = p1;
        pp2 = p2;
        p1 = player1[i];
        p2 = player2[i++];
    }

    return (s1 == s2) ? 0 : s1 > s2 ? 1 : 2;
};

const isTen = (n1, n2) => n1 === 10 || n2 === 10;
